lib/improver/nbhood.py

```diff
@@ -249,8 +361,18 @@ def run(cube, radius_in_km):
             Cube containing the smoothed field after the square neighbourhood
             method has been applied.
         """
+        if isinstance(cube.data, np.ma.MaskedArray):
```

Comment:
That seems reasonable. So long as everything is using either NaNs or masks for MDIs that will be fine. I will add the check for NaNs.